src/serialization.rs

```diff
@@ -287,23 +303,48 @@ impl Serializer {
     where
         V: DeserializeOwned,
     {
+        #[allow(unreachable_patterns)]
         match self.ser_method {
             SerializationMethod::Json => self.json_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "bincode")]
             SerializationMethod::Bin => self.bincode_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "yaml")]
             SerializationMethod::Yaml => self.yaml_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "cbor")]
             SerializationMethod::Cbor => self.cbor_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "json")]
+            _ => self.json_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "bincode")]
+            _ => self.bincode_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "yaml")]
+            _ => self.yaml_serializer.deserialize_data(ser_data),
+            #[cfg(feature = "cbor")]
+            _ => self.cbor_serializer.deserialize_data(ser_data),
```

Comment:
Why is this needed?

Reply:
In my testing, you couldn't individually disable enum fields with features, so all the `SerializationMethod` fields are available. However, the code used with them (for instance the `JsonSerializer` struct) is not generated, so the corresponding match arms can't be generated either without a missing function error. Therefore, to be exhaustive, there have to be catch-all fields for the ones that do exist. Since any combination of features can be used, even turning off the default, all the `SerializationMethod` fields need to have 2 redundant match arms.

Reply:
Got it, that makes sense!